akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -17,6 +17,17 @@ import akka.annotation.{ DoNotInherit, InternalApi }
 abstract class BehaviorInterceptor[O, I] {
   import BehaviorInterceptor._
 
+  /**
+   * Allows for applying the interceptor only to certain message types. Useful if the official protocol and the actual
+   * protocol of an actor causes problems, for example class cast exceptions for a message not of type `O` that
+   * the actor still knows how to deal with. Note that this is only possible to use when `O` and `I` are the same type.
+   *
+   * @return A subtype of `O` that should be intercepted or `null` to intercept all `O`s.
+   *         Subtypes of `O` matching this are passed directly to the inner behavior without interception.
+   */
+  // null for all to avoid having to deal with class tag/explicit class in the default case of no filter
```

Comment:
doesn't a comment here remove the scaladoc/javadoc ?
add it inside the method, as an implementation note

Reply:
No, that's fine (double checked because your question made me less confident). I can move it anyway if you want.